### PR TITLE
solution to dedupe upkeep ids

### PR DESCRIPTION
**Pull request by EasterTheBunny**

### CODE REVIEW

File: internal/keepers/util.go [hunk 1]
```diff
@@ -60,7 +60,11 @@ func (_ *cryptoShuffler[T]) Shuffle(a []T) []T {
 type sortUpkeepKeys []ktypes.UpkeepKey
 
 func (s sortUpkeepKeys) Less(i, j int) bool {
-	return string(s[i]) < string(s[j])
+	// TODO: hacky string split assuming separator
```
Comment (infiloop2) [inline internal/keepers/util.go:63]: can you create a ticket here to fix this.

Also this is same as https://github.com/smartcontractkit/ocr2keepers/blob/main/internal/keepers/filter.go#L80-L81 so would be better to make a common function which does this. (getBlockNumber(upkeepKey), getUpkeepID(upkeepKey))

File: internal/keepers/util.go [hunk 2]
```diff
@@ -138,6 +142,29 @@ func shuffledDedupedKeyList(attributed []types.AttributedObservation, key [16]by
 		return nil, fmt.Errorf("%w: observation dedupe", err)
 	}
 
+	// TODO: another hacky solution assuming upkeep key structure
```
Comment (infiloop2) [inline internal/keepers/util.go:145]: same comment. let's collect all places where we do the split of key into single function
and document the hack in a ticket and link it here

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Add comments about better collecting and documenting the hacks

Also could you add a test for this

File: internal/keepers/util.go (hunk 1 above)
Comment (EasterTheBunny) [reply internal/keepers/util.go:63]: https://smartcontract-it.atlassian.net/browse/AUTO-1480

Comment (infiloop2) [reply internal/keepers/util.go:63]: nit. ideally link them in the code so that anyone who is browsing the repo can get context

Comment (EasterTheBunny) [reply internal/keepers/util.go:63]: dun

Comment (infiloop2) [reply internal/keepers/util.go:63]: todo: remove

File: internal/keepers/util.go [hunk 3]
```diff
@@ -138,6 +143,30 @@ func shuffledDedupedKeyList(attributed []types.AttributedObservation, key [16]by
...
+	// removes duplicate upkeep ids in preferance of ids at higher blocks
+	// needs to be refactored
+	// AUTO-1480
+	sort.Sort(sortUpkeepKeys(keys))
```
Comment (infiloop2) [inline internal/keepers/util.go:150]: rm line 150

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**